### PR TITLE
Feature/touch model when destroy or recover

### PR DESCRIPTION
**Pull request by masylum**
#### :tophat: What? Why?

This is really important for us to understand when a model has been deleted.

### CODE REVIEW

File: lib/immortal.rb [hunk 1]
```diff
@@ -124,17 +124,23 @@ def destroy!
...
-      self.class.unscoped.update_all({ :deleted => false }, "id = #{self.id}")
+      self.class.unscoped.update_all({ deleted: false, updated_at: current_time_from_proper_timezone }, "id = #{self.id}")
       @destroyed = false
       reload
     end
 
+    private
+
+    def current_time_from_proper_timezone
+      ActiveRecord::Base.default_timezone == :utc ? Time.now.utc : Time.now
```
Comment (sauloperez) [inline lib/immortal.rb]: If you want to consider the time zone that was configured in the app, you should use `Time.zone.now`

Comment (somebody32) [inline lib/immortal.rb]: agree on that

Comment (masylum) [inline lib/immortal.rb]: I'm using the exact method that Rails is using to update internally the `updated_at` on doing a touch. Unluckily its `private` not `protected`. See here: https://github.com/rails/rails/blob/master/activerecord/lib/active_record/timestamp.rb#L114

Comment (somebody32) [inline lib/immortal.rb]: OK, i think it is because AR could be used without ActiveSupport

**Comment by somebody32** [comment]
:+1:

**Comment by csaura** [comment]
:+1:

**Pull request merged**